pkg/deploy/deploy.go

```diff
@@ -271,7 +271,8 @@ func manageResource(ctx context.Context, cli client.Client, obj *unstructured.Un
 	if apierrs.IsNotFound(err) {
 		// Set the owner reference for garbage collection
 		// Skip set on CRD, e.g. we should not delete notebook CRD if we delete DSC instance
-		if found.GetKind() != "CustomResourceDefinition" {
+		// Skip on resource if has annotation "opendatahub.io/managed: false"
+		if found.GetKind() != "CustomResourceDefinition" || found.GetAnnotations()["opendatahub.io/managed"] != "false" {
```

Comment:
is the intention here "if it is CustomResourceDefinition OR there is managed=false annotation, then skip setting the reference"? If so should the negation be with AND then?

Reply:
i was not sure in this case should annoation take over CRD kind or CRD kind should always be skipped regardless annoation.
now i think probably the later is more safe, so we do not change behavior, and we can decide what we want later.
i will do the change, what i want to do now:
- if it is CRD regardless annotation, skip
- if it is not CRD and has annoation to false, skip
- othercases, set ownerreference